src/physics/em/detail/SeltzerBergerInteractor.i.hh

```diff
@@ -0,0 +1,145 @@
+ *
+ * See section 10.2.1 of the Geant physics reference 10.6.
+ */
+template<class Engine>
+CELER_FUNCTION Interaction SeltzerBergerInteractor::operator()(Engine& rng)
+{
+    // Allocate space for the bremss photon
+    Secondary* gamma_secondary = this->allocate_(1);
+    if (gamma_secondary == nullptr)
+    {
+        // Failed to allocate space for the secondary
+        return Interaction::from_failure();
+    }
+
+    // Density correction
+    const auto migdal = 4.0 * constants::pi * constants::r_electron
+                        * ipow<2>(constants::lambdabar_electron);
+    real_type density_factor   = material_.electron_density() * migdal;
+    real_type total_energy_val = inc_energy_.value()
+                                 + device_pointers_.electron_mass;
+    real_type density_correction = density_factor * ipow<2>(total_energy_val);
+
+    // Outgoing photon energy sampler
+    SBEnergyDistribution sample_gamma_energy{
+        device_pointers_,
+        inc_energy_,
+        ElementId{0},
+        EnergySq{density_correction},
+        cutoffs_.energy(device_pointers_.ids.gamma)};
+    Energy gamma_exit_energy = sample_gamma_energy(rng);
+
+    // Cross-section scaling for positrons
+    if (particle_id_ == device_pointers_.ids.positron)
+    {
+        SBPositronXsCorrector scale_xs(
+            units::MevMass{device_pointers_.electron_mass},
+            material_.element_view(ElementComponentId{0}),
+            cutoffs_.energy(device_pointers_.ids.gamma),
+            inc_energy_);
+        gamma_exit_energy
+            = Energy{gamma_exit_energy.value() * scale_xs(gamma_exit_energy)};
+    }
+
+    // Construct interaction for change to parent (incoming) particle
+    Interaction result;
+    result.action = Action::spawned;
+    result.energy
+        = units::MevEnergy{inc_energy_.value() - gamma_exit_energy.value()};
+    result.direction               = inc_direction_;
+    result.secondaries             = {gamma_secondary, 1};
+    gamma_secondary[0].particle_id = device_pointers_.ids.gamma;
+
+    // Sample exiting gamma direction
+    UniformRealDistribution<real_type> sample_phi(0, 2 * constants::pi);
+    real_type                          phi = sample_phi(rng);
+    real_type cost = this->sample_cos_theta(inc_energy_, rng);
+    gamma_secondary[0].direction
+        = rotate(from_spherical(cost, phi), inc_direction_);
+
+    // Update parent particle direction
+    for (unsigned int i : range(3))
+    {
+        real_type inc_momentum_i   = inc_momentum_.value() * inc_direction_[i];
+        real_type gamma_momentum_i = result.secondaries[0].energy.value()
+                                     * gamma_secondary[0].direction[i];
+        result.secondaries[0].direction[i] = inc_momentum_i - gamma_momentum_i;
+    }
+
+    return result;
+}
+
+template<class Engine>
+CELER_FUNCTION real_type SeltzerBergerInteractor::sample_cos_theta(
+    const Energy kinetic_energy, Engine& rng)
```

Comment:
As discussed on slack, combine this with `BetheHeitlerInteractor::sample_cos_theta` into a new `physics/em/detail/TsaiUrbanDistribution.hh` or the like.

Reply:
Introduced `TsaiUrbanDistribution`.